packages/muelu/research/regionMG/src/SetupRegionHierarchy_def.hpp

```diff
@@ -1283,13 +1282,17 @@ void vCycle(const int l, ///< ID of current level
           RCP<const Import> rebalanceImporter = coarseSolverData->get<RCP<const Import> >("rebalanceImporter");
 
           RCP<Vector> rebalancedRhs = VectorFactory::Build(rebalanceImporter->getTargetMap());
-          RCP<Vector> rebalancedX = VectorFactory::Build(rebalanceImporter->getTargetMap());
+          RCP<Vector> rebalancedX = VectorFactory::Build(rebalanceImporter->getTargetMap(), true);
```

Comment:
You could cache these vectors.

Reply:
That's true, I don't think we need to do that now if the performance we see on Cori is decent.
The issue with not caching things well probably stems a little bit from the fact that we are not using a "level" like container anywhere in this code and that leads to the catastrophic interface call with 40+ input parameters...
I feel it would be nice if we can add a comment here to remember that issue and tackle it when we refactor the code and introduce a container to simplify this type of things a lot.